src/app/clients/ActionBar.tsx

```diff
@@ -0,0 +1,186 @@
+    "Parcel Collected",
+    "Delivered",
+    "Delivery Failed",
+    "Delivery Cancelled",
+    "Fulfilled with Trussell Trust",
+];
+
+const actions = [
+    "Generate Map",
+    "Print Day Overview",
+    "Print Shopping List",
+    "Print Shipping Label",
+    "Print Driver Overview",
+    "Delete Request",
+];
+
+const OuterDiv = styled.div`
+    display: flex;
+    padding: 1rem;
+    gap: 1rem;
+    border-radius: 0.5rem;
+    background-color: ${(props) => props.theme.surfaceBackgroundColor};
+    margin: 1rem;
+`;
+
+const ActionBar: React.FC<Props> = ({ selected, data }) => {
+    const selectedData = Array.from(selected.map((index) => data[index]));
+
+    const [statusAnchorEl, setStatusAnchorEl] = useState<null | HTMLElement>(null);
+    const [actionAnchorEl, setActionAnchorEl] = useState<null | HTMLElement>(null);
+    const [selectedStatus, setSelectedStatus] = useState<string | null>(null);
+    const [selectedAction, setSelectedAction] = useState<string | null>(null);
+
+    const [statusModal, setStatusModal] = useState(false);
+    const [actionModal, setActionModal] = useState(false);
+
+    const [modalError, setModalError] = useState<string | null>(null);
+
+    const submitStatus = async (date: Dayjs): Promise<void> => {
+        const toInsert = selectedData
+            .map((parcel) => {
+                const event_name = selectedStatus!;
+                const parcel_id = parcel.primary_key;
+                const timestamp = date.set("second", 0).toISOString();
+                return {
+                    event_name,
+                    parcel_id,
+                    timestamp,
+                };
+            })
+            .flat();
+
+        const { error } = await supabase.from("events").insert(toInsert);
+
+        if (error) {
+            setModalError(error.message);
+        } else {
+            setStatusModal(false);
+            setModalError(null);
+        }
+    };
+
+    const submitAction = async (): Promise<void> => {};
+
+    return (
+        <>
+            <ActionBarModal
+                isOpen={statusModal}
+                onClose={() => {
+                    setStatusModal(false);
+                    setModalError(null);
+                }}
+                data={selectedData}
+                status={selectedStatus}
+                header="Apply Status"
+                headerId="status-modal-header"
+                onSubmit={submitStatus}
+                errorText={modalError}
+            />
+            <ActionBarModal
+                isOpen={actionModal}
+                onClose={() => {
+                    setActionModal(false);
+                    setModalError(null);
+                }}
+                data={selectedData}
+                status={selectedAction}
+                header="Perform Action"
+                headerId="action-modal-header"
+                onSubmit={submitAction}
+                errorText={modalError}
+            />
+            <Menu
```

Comment:
Seems like this section and the section below is very similar, would it be better to have a component? or at least have the onClick function separated out.

Reply:
My instinct would suggest that it would be better to keep it inline although other people's comments/opinions are welcome

Reply:
I get what you mean. Happy to keep it as it is!